.github/workflows/ci.yml

```
@@ -0,0 +1,77 @@
+name: CI - Ansible Collection Quality Checks
+
+on:
+  pull_request:
+    branches:
+      - dev
+  push:
+    branches:
+      - dev
+
+jobs:
+  # Sanity tests
+  sanity-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ansible-version: [2.14, 2.16, 2.17]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+
+      - name: Install ansible-core
+        run: |
+          pip install ansible-core==${{ matrix.ansible-version }}
+
+      - name: Prepare collection directory
+        run: |
+          mkdir -p ansible_collections/azure/azcollection
+          rsync -av --exclude ansible_collections --exclude .git ./ ansible_collections/azure/azcollection/
+
+      - name: Run sanity tests
+        working-directory: ansible_collections/azure/azcollection
+        run: |
+          ansible-test sanity --docker --color yes
+
+  # Lint-checks
+  lint-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+
+      - name: Install ansible-lint
+        run: |
+          pip install ansible-lint
+
+      - name: Run ansible-lint
+        run: |
+          ansible-lint .
+
+  # Requirements validation
+  requirements-validation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      
+      - name: Validate requirements.txt
+        run: |
+          echo "Checking requirements.txt for pinned versions..."
+          # Fail if pinned versions exist for non-exempt packages
+          if grep -E '^[a-zA-Z0-9_-]+==[0-9]' requirements.txt | grep -v -E '^(azure-|msgraph-sdk)'; then
```

Can we simplify this to:

```suggestion
          if grep '==' requirements.txt | grep -v -E '^(azure-|msgraph-sdk)'; then
```